motoman_driver/src/io_relay.cpp

```diff
@@ -136,6 +141,40 @@ bool MotomanIORelay::readSingleIoCB(
   return true;
 }
 
+// Service to read a group IO
+bool MotomanIORelay::readGroupIoCB(
+  motoman_msgs::ReadGroupIO::Request &req,
+  motoman_msgs::ReadGroupIO::Response &res)
+{
+  shared_int io_val = -1;
+  std::string err_msg;
+
+  // send message and release mutex as soon as possible
+  this->mutex_.lock();
```

Comment:
This is likely not the correct mutex.

See #389.

(same for the other usages for `mutex_` in this PR)